src/DictVectors/initiators.jl

```diff
@@ -42,302 +78,147 @@ Concrete implementations:
 * [`Initiator`](@ref)
 * [`SimpleInitiator`](@ref)
 * [`CoherentInitiator`](@ref)
+* [`NonInitiator`](@ref)
+
+When defining a new `InitiatorRule`, also define the following:
+
+* [`initiator_valtype`](@ref)
+* [`from_initiator_value`](@ref)
+* [`to_initiator_value`](@ref)
+
+"""
+abstract type InitiatorRule end
+
+"""
+    initiator_valtype(rule::InitiatorRule, T)
 
-When defining a new `InitiatorRule`, also define a corresponding method for [`value`](@ref)!
+Return the `[`AbstractInitiatorValue{T}`](@ref) that is employed by the `rule`.
```

Comment:
I am wondering whether we need such a complicated setup that every `InitiatorRule` is associated with a `AbstractInitiatorValue` type. I.e. would it be possible to incorporate the information in the `InitiatorRule` into the type of the values, and then dispatch on that?

Reply:
I have considered this, but decided against it because then changing the initiator threshold changes the type of the vector.